### DISCUSSION TITLE
Service Worker in SPA mode

### DISCUSSION THREAD

**Discussion by yethie** [quasarframework/quasar#11942] [General - Components / Directives / etc] [answered]
Hi!
I would like to use a service worker in a SPA application, does Quasar provide a feature for service workers in SPAs like it does for PWAs?

**Comment by yusufkandemir** [comment, marked as answer]
Hi, you can just use PWA **build mode** instead. PWA **build mode** is _basically_ SPA **build mode** + a service worker + a web app manifest. You don't necessarily need to make use of all possible PWA features or adopt its best practices; if that's why you are trying to avoid it.

**Reply by yethie** [reply]
Hello, thanks for your answer! When I use the PWA build mode and load the site in the browser (Chrome in my tests) I get an "Install  [Project Name]" icon in the address bar. Users seeing the icon may think that the app may be used offline, while it is not the case. I just would like to prevent the browser from showing that icon.

**Reply by yusufkandemir** [reply]
A PWA doesn't necessarily need to work offline. So, I think you don't need to worry too much about it, you can just leave it be. But, if you just don't want it, or you know your user base will definitely be confused, then you can listen to the `beforeinstallprompt` event and prevent it. See: https://web.dev/promote-install/#browser-promotion

**Comment by nachoalcazar** [comment]
If I build my application with:
- quasar mode add pwa
- quasar dev -m pwa

,,, to use service-worker... How Can I compile and build the applitacion to use this service-worker in electron/capacitor mode?

Greetings
